framework/src/main/java/org/checkerframework/common/basetype/BaseTypeVisitor.java

```diff
@@ -3500,6 +3500,27 @@ protected void checkEnclosingExpr(NewClassTree node, AnnotatedExecutableType con
                 argumentReceiverType = atypeFactory.getAnnotatedType(node.getEnclosingExpression());
             } else {
                 argumentReceiverType = atypeFactory.getReceiverType(node);
+                /*
+                  Consider the following code:
+                  Class Outer {
+                      Class Inner{}
+                  }
+                  Class Top{
+                      void test(Outer outer) {
+                          outer.new Inner(){};
+                      }
+                  }
+                  In Java versions below 11, the argumentReceiverType of outer.new Inner(){} is Top instead of Outer, because java8 organizes
+                  newClassTree in a different way: there is a synthetic argument. Hence, use this synthetic argument when the underlying types are different.
```

Comment:
"Synthetic argument" for what?

Reply:
Thanks! Reworded the comment in my local.